src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2941,6 +2941,14 @@ private bool isExactType(CORINFO_CLASS_STRUCT_* cls)
             return _compilation.IsEffectivelySealed(type);
         }
 
+        private TypeCompareState isNullableType(CORINFO_CLASS_STRUCT_* cls)
+        {
+            TypeDesc type = HandleToObject(cls);
+            Debug.Assert(!type.IsGenericParameter);
```

Comment:
I don't think we need the assert. I'm only running the leg with an assert to double check. It's a nonsensical value, but I could come up with dozen(s) of other non-sensical values we could assert here. It feels like asserting it's not a generic parameter is just confusing (this should never be a generic parameter in any of the APIs we have here and then why we only assert here?).